addons/sale/models/sale.py

```diff
@@ -631,28 +631,62 @@ def _create_invoices(self, grouped=False, final=False, date=None):
 
         # 1) Create invoices.
         invoice_vals_list = []
+        invoice_item_sequence = 0
         for order in self:
-            pending_section = None
+            prepared_pending_section = None
+            down_payments = []
 
             # Invoice values.
             invoice_vals = order._prepare_invoice()
 
             # Invoice line values (keep only necessary sections).
+            invoice_line_ids = []
             for line in order.order_line:
                 if line.display_type == 'line_section':
-                    pending_section = line
+                    prepared_pending_section = line._prepare_invoice_line()
```

Comment:
What about adding an optional line_vals kwargs parameter to _prepare_invoice_line that, if defined, is added to the returned values ? Like
```
def _prepare_invoice_line(self, **optional_values):
    res = {}
    ...
    if optional_values:
        res.update(optional_values)
    return res
```

That way you can do line._prepare_invoice_line(sequence=invoice_item_sequence) when necessary.

What do you think ?

Reply:
Good idea, implemented!